ntcore/src/main/native/include/networktables/NetworkTableInstance.inc

```diff
@@ -241,4 +241,36 @@ inline NT_Listener NetworkTableInstance::AddLogger(unsigned int min_level,
   return ::nt::AddLogger(m_handle, min_level, max_level, std::move(func));
 }
 
+inline bool NetworkTableInstance::HasSchema(std::string_view name) {
+  return ::nt::HasSchema(m_handle, name);
+}
+
+inline void NetworkTableInstance::AddSchema(std::string_view name,
+                                            std::string_view type,
+                                            std::span<const uint8_t> schema) {
+  ::nt::AddSchema(m_handle, name, type, schema);
+}
+
+inline void NetworkTableInstance::AddSchema(std::string_view name,
+                                            std::string_view type,
+                                            std::string_view schema) {
+  ::nt::AddSchema(m_handle, name, type, schema);
+}
+
+template <wpi::ProtobufSerializable T>
+void NetworkTableInstance::AddProtobufSchema(wpi::ProtobufMessage<T>& msg) {
+  msg.ForEachProtobufSchema(
+      [this](auto typeString) { return !HasSchema(typeString); },
```

Comment:
Do we want to handle the (unlikely) case where a protobuf schema gets added without all of its descendants? (If that happens, then `ForEachProtobufSchema` will see that the schema is not wanted and skip its descendants, even though some were not already added) Checking each one individually would definitely degrade performance, but I just wanted to double-check that this behavior is fine (especially since adding just a single schema is publicly exposed).

Reply:
It feels pretty unlikely someone will add a protobuf schema using the single schema method, as it's relatively complex to do so.

There's a bigger issue with protobuf schemas I'm trying to figure out right now. I'm not sure it's actually useful to serialize just the DescriptorProto. The dynamic stuff in protobuf seems to be designed around FileDescriptorProto instead, and it's unclear if we can do what we want without that.

Reply:
I've switched C++ to publishing FileDescriptorProto.  The Java version is blocked on QuickBuffers adding support.